app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,177 @@
+// See LICENSE.txt for license information.
+
+import {Model, Q} from '@nozbe/watermelondb';
+import {DeviceEventEmitter} from 'react-native';
+
+import {storeMyChannelsForTeam, markChannelAsUnread, markChannelAsViewed, updateLastPostAt} from '@actions/local/channel';
+import {markPostAsDeleted} from '@actions/local/post';
+import {fetchMyChannel, markChannelAsRead} from '@actions/remote/channel';
+import {fetchPostAuthors, fetchPostById} from '@actions/remote/post';
+import {ActionType, Events} from '@constants';
+import {MM_TABLES} from '@constants/database';
+import DatabaseManager from '@database/manager';
+import {queryMyChannel} from '@queries/servers/channel';
+import {queryPostById} from '@queries/servers/post';
+import {queryCurrentChannelId, queryCurrentUserId} from '@queries/servers/system';
+import {isFromWebhook, isSystemMessage, shouldIgnorePost} from '@utils/post';
+
+import type {WebSocketMessage} from '@typings/api/websocket';
+
+const {SERVER: {POST}} = MM_TABLES;
+
+export async function handleNewPostEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const post: Post = JSON.parse(msg.data.post);
+    const currentUserId = await queryCurrentUserId(database.database);
+
+    const existing = await queryPostById(database.database, post.pending_post_id);
+
+    if (existing) {
+        return;
+    }
+
+    database.operator.handlePosts({
+        actionType: ActionType.POSTS.RECEIVED_NEW,
+        order: [post.id],
+        posts: [post],
+    });
+
+    const models: Model[] = [];
+
+    // Ensure the channel membership
+    let myChannel = await queryMyChannel(database.database, post.channel_id);
+    if (!myChannel) {
+        const myChannelRequest = await fetchMyChannel(serverUrl, '', post.channel_id, true);
+        if (myChannelRequest.error) {
+            return;
+        }
+
+        // We want to have this on the database so we can make any needed update later
+        const myChannelModels = await storeMyChannelsForTeam(serverUrl, '', myChannelRequest.channels!, myChannelRequest.memberships!, false);
+        if (myChannelModels.error) {
+            return;
+        }
+
+        myChannel = await queryMyChannel(database.database, post.channel_id);
+        if (!myChannel) {
+            return;
+        }
+    }
+
+    // If we don't have the root post for this post, fetch it from the server
+    if (post.root_id) {
+        const rootPost = await queryPostById(database.database, post.root_id);
+
+        if (!rootPost) {
+            fetchPostById(serverUrl, post.root_id);
+        }
+    }
+
+    const currentChannelId = await queryCurrentChannelId(database.database);
+
+    if (post.channel_id === currentChannelId) {
+        const data = {
+            channelId: post.channel_id,
+            rootId: post.root_id,
+            userId: post.user_id,
+            username: '',
```

Comment:
do you need this? or with the user_id is enough? haven't completed the review on #5844

Reply:
For stop, username should not be needed. I added just for the sake of completion. I will remove it.